tdp/cli/commands/vars_edit.py

```diff
@@ -0,0 +1,150 @@
+    database_dsn: str,
+    validate: bool,
+    vars: Path,
+):
+    """Edit a variables file for a service.
+
+    SERVICE_NAME: Name of the service to edit variables for.
+    SERVICE_COMPONENT_PARAMETER: Name or path of the component to edit variables for. If not specified, edit the service variables file.
+
+    \b
+    Examples:
+        tdp edit-vars hdfs
+        tdp edit-vars hdfs datanode --commit_message "updated datanode variables"
+        tdp edit-vars hdfs hdfs_datanode.yml
+    """
+    cluster_variables = ClusterVariables.get_cluster_variables(
+        collections, vars, validate=validate
+    )
+
+    # Check if service exists
+    if service_name not in cluster_variables:
+        strErr = f"Error unknown service '{service_name}'"
+        logger.error(strErr)
+        raise click.ClickException(strErr)
+
+    service_variables = cluster_variables[service_name]
+    repo = service_variables.repository
+
+    # Check if service is clean
+    if not service_variables.clean:
+        raise click.ClickException(
+            f"Error service '{service_name}' is not clean. Commit or stash your changes before editing variables."
+        )
+
+    # Get the variable file to edit
+    base_path = vars / service_name
+    if service_component_parameter is None:
+        # tdp edit-vars service
+        variables_file = base_path / (service_name + YML_EXTENSION)
+    elif service_component_parameter.endswith(YML_EXTENSION):
+        # tdp edit-vars service service.yml OR tdp edit-vars service service_component.yml
+        variables_file = base_path / service_component_parameter
+    else:
+        # tdp edit-vars service component
+        variables_file = base_path / (
+            service_name + "_" + service_component_parameter + YML_EXTENSION
+        )
+
+    # Check if component exists
+    service_component_name = ServiceComponentName.from_full_name(variables_file.stem)
+    if not service_component_name.is_service:
+        components_of_service = collections.get_components_from_service(service_name)
+        if service_component_name not in components_of_service:
+            raise click.ClickException(
+                f"Error unknown component '{service_component_name.component_name}' for service '{service_component_name.service_name}'"
+            )
+
+    logger.debug(f"Editing {variables_file.name} for service {service_name}")
+
+    # Loop until variables file format has no errors or user aborts editing file
+    while True:
+        click.edit(
+            filename=str(variables_file),
+        )
+
+        # Pause until user press any key
+        value: str = click.prompt(
+            "Press any key when done editing ('q' to cancel)",
+            type=str,
+            prompt_suffix="",
+            default="continue",
+            show_default=False,
+        )
+        if value.lower() == "q":
+            repo.restore_file(str(variables_file))
+            click.ClickException("Aborted changes")
```

Comment:
I think we have to add ```raise``` : the program goes on even if we choose 'q' ^^'

Reply:
Yes thx. I fixed it!